### PR TITLE
Change base logger to Serilog

### PR DESCRIPTION
**Pull request by AkulovOleg**
## Description
We need to have possibilities for using different log providers. In our case we need Grafana.Loki.
So it's better to have **Serilog** for adding in partners' projects any providers which they need.

1) I didn't test the platform with ApplicationInsights, so please test it.
2) After using this library when can just add the needed Sink package to any of our custom modules and set up sink in appsettings.json file.

Description of the library: https://github.com/serilog/serilog
Here you can find available providers: https://github.com/serilog/serilog/wiki/Provided-Sinks

### CODE REVIEW

File: src/VirtoCommerce.Platform.Web/Program.cs [hunk 1]
```diff
@@ -20,13 +21,11 @@ public static void Main(string[] args)
            Host.CreateDefaultBuilder(args)
               .ConfigureLogging((hostingContext, logging) =>
               {
-                  logging.AddConfiguration(hostingContext.Configuration.GetSection("Logging"));
-                  logging.AddConsole();
-                  logging.AddDebug();
-                  logging.AddEventSourceLogger();
```
Comment (AkulovOleg) [inline src/VirtoCommerce.Platform.Web/Program.cs:26]: Do you still need logging of event sources?

File: src/VirtoCommerce.Platform.Web/VirtoCommerce.Platform.Web.csproj [hunk 2]
```diff
@@ -40,6 +40,10 @@
         <PackageReference Include="Microsoft.VisualStudio.Web.BrowserLink" Version="2.2.0" />
         <PackageReference Include="OpenIddict.AspNetCore" Version="3.0.4" />
         <PackageReference Include="OpenIddict.EntityFrameworkCore" Version="3.0.4" />
+        <PackageReference Include="Serilog.AspNetCore" Version="6.1.0" />
+        <PackageReference Include="Serilog.Sinks.ApplicationInsights" Version="4.0.0" />
+        <PackageReference Include="Serilog.Sinks.Console" Version="4.1.0" />
+        <PackageReference Include="Serilog.Sinks.Debug" Version="2.0.0" />
```
Comment (ksavosteev) [inline src/VirtoCommerce.Platform.Web/VirtoCommerce.Platform.Web.csproj:46]: Add `<PackageReference Include="Serilog.Extensions.Hosting" Version="5.0.1" />`.

**Review by ksavosteev — CHANGES_REQUESTED** [review]

File: src/VirtoCommerce.Platform.Web/VirtoCommerce.Platform.Web.csproj (hunk 2 above)
Comment (AkulovOleg) [reply src/VirtoCommerce.Platform.Web/VirtoCommerce.Platform.Web.csproj:46]: added

File: src/VirtoCommerce.Platform.Web/Program.cs [hunk 3]
```diff
@@ -20,13 +21,11 @@ public static void Main(string[] args)
...
-                  //Enable Azure logging
-                  //https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-2.2#logging-in-azure
-                  logging.AddAzureWebAppDiagnostics();
+                  logging.ClearProviders();
+              })
+              .UseSerilog((context, loggerConfiguration) =>
```
[the remark this answers is not in the document]
Comment (AkulovOleg) [reply src/VirtoCommerce.Platform.Web/Program.cs]: changed

File: src/VirtoCommerce.Platform.Web/appsettings.Development.json [hunk 4]
```diff
@@ -1,12 +1,18 @@
...
-            "Microsoft.AspNetCore.SignalR": "Trace",
-            "Microsoft.AspNetCore.Http.Connections": "Trace"
-        }
-    }
+  "Serilog": {
+    "Using": [
+      "Serilog.Sinks.Console"
+    ],
+    "MinimumLevel": {
+      "Default": "Trace",
```
[the remark this answers is not in the document]
Comment (AkulovOleg) [reply src/VirtoCommerce.Platform.Web/appsettings.Development.json]: changed

**Comment by OlegoO** [comment]
Replaced with 
https://github.com/VirtoCommerce/vc-platform/pull/2644
https://github.com/VirtoCommerce/vc-module-app-insights

**Pull request closed without merging**